test/FunctionalTests/Microsoft.OData.Core.Tests/Json/JsonWriterAsyncTests.cs

```diff
@@ -383,6 +383,39 @@ private async Task VerifyWriterPrimitiveValueWithIeee754CompatibleAsync<T>(T par
             Assert.Equal(expected, this.builder.ToString());
         }
 
+        [Theory]
+        [InlineData(124.45, "124.45")]
+        // We write the .0 for doubles without a fractional part
+        // for consistency with the original JsonWriter implementation
+        // and with previous versions. However, it's not a hard requirement.
+        // Clients should not rely on the .0 to decide whether a value
+        // is an integer or double.
+        // In the future we can consider relaxing the need to add a .0 (possibly behind a feature flag).

```

Comment:
Maybe we should create an issue to streamline this in ODL 9. The issue would also serve to document our intentions...

Reply:
We didn't make this change in 7.x.We'll make it in 8.x.